### DISCUSSION TITLE
MenuItem HotKey/InputGesture doesn't work until Menu shown at least once (v11)

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#12064] [Q&A]
I notice an annoying behavior where none of the hotkeys works until Menu is open for the first time.

Menu:
![image](https://github.com/AvaloniaUI/Avalonia/assets/113281/6f7c92d5-5a13-4d63-8d29-d450cee746f6)

Problem:

1. Open App
2. Press any hotkey
3. Nothing happens

How to "fix"

1. Open App
2. Click on the menu and then close it
3. Press any hotkey
4. It works as expected

Note that for the hotkey to work it parent menu should be open first.
Not sure if this is an issue with the framework

My usage:

```xml
<Menu DockPanel.Dock="Top">
				<MenuItem Name="MainMenuFile" Header="_File">
					<MenuItem Name="MainMenuFileOpen"
							  Header="_Open"
							  HotKey="Ctrl+O" InputGesture="Ctrl+O"
							  Command="{Binding MenuFileOpenClicked}"
							  i:MenuItem.Icon="fa-solid fa-file-import"/>
                                </MenuItem>
</Menu>
```

Note: Works as expected on v0.10
Additional context: If `DataContext` is changed, the hotkeys no longer work and require same open menu "hack"

**Comment by jp2masa** [comment]
#2441
